File: 2 Sorting/C Insertion Sort/insertion-sort.py
def insertion_sort(arr):
  n = len(arr)
  for i in range(1, n):
    key = arr[i]
    j = i - 1
    while j >= 0 and arr[j] > key:
      arr[j + 1] = arr[j]
      j -= 1
    arr[j + 1] = key

arr = [2,5,5,7,3,-5,6]
# [-5, 2, 3, 5, 5, 6, 7]
insertion_sort(arr)
print(arr)

File: Sorting/Insertion Sort/insertion-sort.cpp
#include <bits/stdc++.h>
using namespace std;

void insertionSort(int arr[], int n)
{
  for (int i = 1; i < n; i++)
  {
    int j = i - 1;
    int key = arr[i];
    while (j >= 0 && arr[j] > key)
    {
      arr[j + 1] = arr[j];
      j--;
    }
    arr[j + 1] = key;
  }
}

main()
{
  int n;
  cin >> n;
  int arr[n];
  for (int i = 0; i < n; i++)
    cin >> arr[i];

  insertionSort(arr, n);

  for (auto x : arr)
    cout << x << " ";
}